src/Mod/Sketcher/Gui/Utils.cpp

```diff
@@ -355,7 +355,7 @@ bool SketcherGui::isBsplineKnotOrEndPoint(
 
     const Part::Geometry* geo = Obj->getGeometry(GeoId);
     // end points of B-Splines are also knots
-    if (geo->is<Part::GeomBSplineCurve>()
+    if (geo && geo->is<Part::GeomBSplineCurve>()
```

Comment:
This is the core fix. So what happens is that if you have a single circle, select it, the selection is now : 
object = Sketch
Subname = Edge1
The undo does not change the selection! So it stays this Sketch.Edge 1.
But the circle is not here so `getGeometry(GeoId)` returns nullptr. And then crash.

So this check is definitely needed. It should be merged.

But it does not solve the underlying problem : Undo does not clear selection. So even if it does not crash anymore, the selection is still `Sketch.Edge1`. Hence the error messages.

If you have more than 1 edge in your sketch, the edge2 becomes edge1 because of the undo, and becomes selected.